src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -128,31 +128,27 @@ public boolean unlock(UUID snapshotId, String flightId) {
     public UUID createAndLock(Snapshot snapshot, String flightId) {
         logger.debug("createAndLock snapshot " + snapshot.getName());
 
-        String sql = "INSERT INTO snapshot (name, description, profile_id, flightid) " +
-            "VALUES (:name, :description, :profile_id, :flightid) ";
+        String sql = "INSERT INTO snapshot (name, description, profile_id, id, flightid) " +
+            "VALUES (:name, :description, :profile_id, :id, :flightid) ";
         MapSqlParameterSource params = new MapSqlParameterSource()
             .addValue("name", snapshot.getName())
             .addValue("description", snapshot.getDescription())
             .addValue("profile_id", snapshot.getProfileId())
+            .addValue("id", snapshot.getId())
             .addValue("flightid", flightId);
-        DaoKeyHolder keyHolder = new DaoKeyHolder();
         try {
-            jdbcTemplate.update(sql, params, keyHolder);
+            jdbcTemplate.update(sql, params);
         } catch (DuplicateKeyException dkEx) {
-            throw new InvalidSnapshotException("Snapshot name already exists: " + snapshot.getName(), dkEx);
+            throw new InvalidSnapshotException(
+                "Snapshot name or id already exists: " + snapshot.getName() + ", " + snapshot.getId(), dkEx);
         }
 
-        UUID snapshotId = keyHolder.getId();
-        snapshot
-            .id(snapshotId)
-            .createdDate(keyHolder.getCreatedDate());
-
-        snapshotTableDao.createTables(snapshotId, snapshot.getTables());
+        snapshotTableDao.createTables(snapshot.getId(), snapshot.getTables());
         for (SnapshotSource snapshotSource : snapshot.getSnapshotSources()) {
             createSnapshotSource(snapshotSource);
         }
 
-        return snapshotId;
+        return snapshot.getId(); // TODO should this return nothing now?
```

Comment:
I would remove it. It seems odd to have a method return one of its inputs to the caller.